Remove RSAES-pkcs1-v1.5 support

nmahendru opened the pull request:
#### 31e806507852348d649add0eab1363d9a1fdab12
<pre>
Remove RSAES-pkcs1-v1.5 support
<a href="https://bugs.webkit.org/show_bug.cgi?id=281144">https://bugs.webkit.org/show_bug.cgi?id=281144</a>
<a href="https://rdar.apple.com/137603955">rdar://137603955</a>

Reviewed by David Kilzer and Youenn Fablet.

It is already deprecated in chrome and firefox and we have had it disabled by default for some
time now. See: <a href="https://commits.webkit.org/259018@main">https://commits.webkit.org/259018@main</a>

It should be safe to just remove it now as we haven&apos;t heard any claims.
It&apos;s an algorithm with a long history of issues and should not be used.
more details:
<a href="https://datatracker.ietf.org/doc/html/rfc8017#section-7.2">https://datatracker.ietf.org/doc/html/rfc8017#section-7.2</a>

* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key.html: Removed.
* LayoutTests/crypto/workers/subtle/rsa-postMessage-worker-expected.txt: Removed.
* LayoutTests/crypto/workers/subtle/rsa-postMessage-worker.html: Removed.
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-decrypt-expected.txt: Removed.
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html: Removed.
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-encrypt-expected.txt: Removed.
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):
(WebCore::isRSAESPKCSWebCryptoDeprecated): Deleted.
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSAES_PKCS1_v1_5.cpp: Removed.
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSAES_PKCS1_v1_5.h: Removed.
* Source/WebCore/crypto/cocoa/CryptoAlgorithmRSAES_PKCS1_v1_5Mac.cpp: Removed.
* Source/WebCore/crypto/cocoa/CryptoAlgorithmRegistryMac.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSAES_PKCS1_v1_5GCrypt.cpp: Removed.
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSAES_PKCS1_v1_5OpenSSL.cpp: Removed.
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/platform/OpenSSL.cmake:
* Source/WebCore/platform/SourcesGCrypt.txt:

Canonical link: <a href="https://commits.webkit.org/285703@main">https://commits.webkit.org/285703@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b0a6ea8536f4f6a59f7539a01d1647907dcc395c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73482 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52911 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26290 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/77747 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/24720 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/75597 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62042 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/696 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/57742 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16180 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76549 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47820 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63221 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38152 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/44432 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/20706 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23051 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/66617 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66235 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21057 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79368 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/72739 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/798 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/288 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66140 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/939 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63234 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65420 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9269 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7447 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/94520 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11334 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/761 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/3504 "Built successfully") | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/20775 "Found 2 new JSC stress test failures: stress/json-stringify-inspector-check.js.no-llint, wasm.yaml/wasm/stress/cc-int-to-int-cross-module-with-exception.js.default-wasm (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/791 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/777 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/797 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/crypto/SubtleCrypto.cpp:
```diff
@@ -153,10 +147,7 @@ static ExceptionOr<std::unique_ptr<CryptoAlgorithmParameters>> normalizeCryptoAl
-                return Exception { ExceptionCode::NotSupportedError, "RSAES-PKCS1-v1_5 support is deprecated"_s };
-            result = makeUnique<CryptoAlgorithmParameters>(params.releaseReturnValue());
-            break;
+            return Exception { ExceptionCode::NotSupportedError, "RSAES-PKCS1-v1_5 support is deprecated"_s };
```
Should we even keep this, and remove CryptoAlgorithmIdentifier::RSAES_PKCS1_v1_5 as well?

nmahendru replied:
Trying to keep this as this is one way to give an exception similar to Chrome and Firefox if someone tries to use this algorithm.

youennf reviewed (approved):
LGTM once bots are happy, but please see the comment related to removing the enum value as well.

ddkilzer reviewed (approved):
r=me

nmahendru committed:
Remove RSAES-pkcs1-v1.5 support
https://bugs.webkit.org/show_bug.cgi?id=281144
rdar://137603955

Reviewed by David Kilzer and Youenn Fablet.

It is already deprecated in chrome and firefox and we have had it disabled by default for some
time now. See: https://commits.webkit.org/259018@main

It should be safe to just remove it now as we haven't heard any claims.
It's an algorithm with a long history of issues and should not be used.
more details:
https://datatracker.ietf.org/doc/html/rfc8017#section-7.2

* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key.html: Removed.
* LayoutTests/crypto/workers/subtle/rsa-postMessage-worker-expected.txt: Removed.
* LayoutTests/crypto/workers/subtle/rsa-postMessage-worker.html: Removed.
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-decrypt-expected.txt: Removed.
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html: Removed.
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-encrypt-expected.txt: Removed.
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):
(WebCore::isRSAESPKCSWebCryptoDeprecated): Deleted.
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSAES_PKCS1_v1_5.cpp: Removed.
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSAES_PKCS1_v1_5.h: Removed.
* Source/WebCore/crypto/cocoa/CryptoAlgorithmRSAES_PKCS1_v1_5Mac.cpp: Removed.
* Source/WebCore/crypto/cocoa/CryptoAlgorithmRegistryMac.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSAES_PKCS1_v1_5GCrypt.cpp: Removed.
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSAES_PKCS1_v1_5OpenSSL.cpp: Removed.
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/platform/OpenSSL.cmake:
* Source/WebCore/platform/SourcesGCrypt.txt:

Canonical link: https://commits.webkit.org/285703@main
